h2o-py/tests/testdir_misc/pyunit_log4j_cve.py

```diff
@@ -0,0 +1,84 @@
+import sys
+sys.path.insert(1,"../../")
+import socket
+import h2o
+from tests import pyunit_utils
+import time
+import threading
+
+
+def malicious_log():
+    print("Waiting 3s before sending malicious logging call ${jndi:ldap://127.0.0.1:50001/}")
+    time.sleep(3)
+    print("Calling log_and_echo")
+    h2o.log_and_echo('${jndi:ldap://127.0.0.1:50001/}')
+    print("Finished log_and_echo")
+    time.sleep(3)
+    print("Sending poison pill to close the connection")
+    send_poison_pill()
+
+
+def send_poison_pill():
+    sock = None
+    for res in socket.getaddrinfo("localhost", 50001, socket.AF_INET, socket.SOCK_STREAM):
+        af, socktype, proto, canonname, sa = res
+        try:
+            sock = socket.socket(af, socktype, proto)
+        except OSError as oe:
+            print(oe)
+            sock = None
+            continue
+        try:
+            sock.connect(sa)
+        except OSError as oe:
+            sock.close()
+            print(oe)
+            sock = None
+            continue
+        break
+    if sock is None:
+        return
+    try:
+        with sock:
+            sock.close()
+    except OSError as oe:
+        print(oe)
+
+
+def test_log4j_cve():
+    sock = None
+    for res in socket.getaddrinfo("localhost", 50001, socket.AF_INET,
+                                  socket.SOCK_STREAM, 0, socket.AI_PASSIVE):
+        af, socktype, proto, canonname, sa = res
+        try:
+            sock = socket.socket(af, socktype, proto)
+        except OSError:
+            sock = None
+            continue
+        try:
+            sock.bind(sa)
+            sock.listen(1)
+        except OSError:
+            sock.close()
+            sock = None
+            continue
+        break
+    assert sock is not None
+
+    t = threading.Thread(target=malicious_log)
+    t.start()
+
+    conn, _ = sock.accept()
+    with conn:
+        data = conn.recv(10)
+        # if we do receive any data - it was H2O's log4j that called us
+        called_by_log4j = len(data) > 0
+        conn.close()
+
+    assert not called_by_log4j
```

Comment:
I would wrap this with 
```python
try:
    …
finally:
    sock.close()
```
otherwise, on failing test, the socket remains open

Reply:
and it makes me doubt about the utility of the poison pill

Reply:
I see, the poison pill is indeed useful on a fixed jar :)

Reply:
FYI, a simpler test, without the poison pill, but relying on timeout when waiting for the incoming connection:
```python
def test_log4j_cve():
    sock = None
    for res in socket.getaddrinfo("localhost", 50001, socket.AF_INET,
                                  socket.SOCK_STREAM, 0, socket.AI_PASSIVE):
        af, socktype, proto, canonname, sa = res
        try:
            sock = socket.socket(af, socktype, proto)
        except OSError:
            sock = None
            continue
        try:
            sock.bind(sa)
            sock.settimeout(5)
            sock.listen(1)
        except OSError:
            sock.close()
            sock = None
            continue
        break
    assert sock is not None

    try:
        t = threading.Thread(target=malicious_log)
        t.start()

        conn, _ = sock.accept()
        with conn:
            data = conn.recv(10)
            print(data.decode('utf-8'))
            # if we do receive any data - it was H2O's log4j that called us
            called_by_log4j = len(data) > 0
            conn.close()
            
        assert not called_by_log4j
    except socket.timeout:
        pass
    finally:
        sock.close()
```